build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.java.gradle.kts

```diff
@@ -54,9 +54,19 @@ val internal: Configuration =
 dependencies { "internal"(platform("com.hedera.hashgraph:hedera-dependency-versions")) }
 
 sourceSets.all {
-    configurations.getByName(annotationProcessorConfigurationName) { extendsFrom(internal) }
     configurations.getByName(compileClasspathConfigurationName) { extendsFrom(internal) }
     configurations.getByName(runtimeClasspathConfigurationName) { extendsFrom(internal) }
+
+    dependencies {
+        // For dependencies of annotation processors use versions from 'hedera-dependency-versions',
+        // but not 'runtime' dependencies of the platform (JAVA_API instead of JAVA_RUNTIME).
+        annotationProcessorConfigurationName("com.hedera.hashgraph:hedera-dependency-versions") {
+            attributes {
+                attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.JAVA_API))
+                attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.REGULAR_PLATFORM))
+            }
+        }
+    }
```

Comment:
These adjustments for how the "dependency platform" (hedera-dependency-versions) is used for annotation processing and protobuf generation are necessary as we not also use it as a central place to define runtime only dependencies for test/application execution.
We do not want those dependencies on the path of the annotation processors or protobufs generator.